### DISCUSSION TITLE
Shared library?

### DISCUSSION THREAD

**Discussion by fridrich** [AlbrechtL/welle.io#821] [Ideas]
I had this idea of shared library between the welle-io and welle-cli that would eventually allow hooking to the welle another interface. I am planning to work on one that would work on LCD and small OLED displays. I have some proof-of concept code, but first, I would like to know whether this is not something that will meet hard opposition before I get deep into it.

A proof-of-concept patch is https://paste.opensuse.org/pastes/ea8c8d8cbfc6

The question is what sources one wants there and what headers installed. But the principle is all there and it works for me.

**Comment by fridrich** [comment]
Also when working on this I realized that the version returned by `welle-io -v` and the one returned by `welle-cli -v` are different strings. It would be good to sync them to one of them.

**Comment by mpbraendli** [comment]
Oh, that patch is enough?! (it only touches the CMakeLists, not code, correct?)

I was expecting some issues due to the actual interface used between frontends and receiver backend, but apparently that not the case.

**Comment by fridrich** [comment]
Ok, I just took the sources that were outside the welle-gui and welle-cli and that both use. That somehow abstracts the backends and some of the various, but also the mpg123 and faad for the while, althought welle-cli does not need it.
But as I said, the sources list is a topic to discuss.

In the meantime I made another PR where I basically change the way the index.html and index.js are served. That version converts them during the build to two *.h files that I include in the web*interface*.cpp and I changed the prototype of the send_file and it works like a charm here.

I am trying to come up with a solution where an interface could be hooked on the library and for instance have some 20x4 LDC display and a rotary encoder as display/control device. That would make for a nice Pi02 DAB radio :)

**Comment by fridrich** [comment]
Also one would need to see with windows whether it is able to produce a dll with sane exports. Just to be sure we are not interposing something in ELF, which would not work in PE.
